### PR TITLE
Enable mTLS with CF API

### PR DESCRIPTION
**Pull request by AP-Hunt**
# Overview
Addresses #35: Enables `vault-plugin-auth-cf` to use mutual TLS when communicating with the Cloud Foundry API. Documents how to configure this, and notes this to be careful around. This change should have no impact on the function of the plugin.

# Design of Change
We added 2 new configuration parameters (certificate and key) to the plugin. If they're set, the plugin uses them to form a key & cert pair, which it then uses to perform mutual TLS with the configured Cloud Foundry API.

# Related Issues/Pull Requests
[x] [Issue #35](https://github.com/hashicorp/vault-plugin-auth-cf/issues/35)

# Contributor Checklist
[x] Add relevant docs to upstream Vault repository, or sufficient reasoning why docs won’t be added yet
[vault#10356](https://github.com/hashicorp/vault/pull/10356)
[N/A] Add output for any tests not ran in CI to the PR description (eg, acceptance tests)
[x] Backwards compatible; config parameters are ignored if they're not set

### CODE REVIEW

File: path_config.go [hunk 1]
```diff
@@ -34,6 +34,22 @@ func (b *backend) pathConfig() *framework.Path {
 				},
 				Description: "The PEM-format CA certificates that are acceptable for the CF API to present.",
 			},
+			"cf_api_mutual_tls_certificate": {
+				Type: framework.TypeString,
+				DisplayAttrs: &framework.DisplayAttributes{
+					Name:  "CF API Mutual TLS Certificate",
+					Value: `-----BEGIN CERTIFICATE----- ... -----END CERTIFICATE-----`,
+				},
+				Description: "The PEM-format certificates that is presented for mutual TLS with the CloudFoundry API. If not set, mutual TLS is not used",
```
Comment (jasonodonnell) [inline path_config.go:43]:
```suggestion
				Description: "The PEM-format certificates that are presented for mutual TLS with the CloudFoundry API. If not set, mutual TLS is not used",
```

File: path_config.go [hunk 2]
```diff
@@ -34,6 +34,22 @@ func (b *backend) pathConfig() *framework.Path {
...
+			},
+			"cf_api_mutual_tls_key": {
+				Type: framework.TypeString,
+				DisplayAttrs: &framework.DisplayAttributes{
+					Name:  "CF API Mutual TLS Key",
+					Value: `-----BEGIN RSA PRIVATE KEY----- ... -----END RSA PRIVATE KEY-----`,
+				},
+				Description: "The PEM-format private key that is used for mutual TLS with the CloudFoundry API. If not set, mutual TLS is not used",
```
Comment (jasonodonnell) [inline path_config.go:51]:
```suggestion
				Description: "The PEM-format private key that are used for mutual TLS with the CloudFoundry API. If not set, mutual TLS is not used",
```

**Review by jasonodonnell — APPROVED** [review]
This looks good to me! Just minor things to clean up. We'll expect this to be released in Vault 1.7.

**Comment by AP-Hunt** [comment]
Agree with your suggestions. I've amended the commit and pushed the branch up.

**Pull request merged**